### PR TITLE
Introduce OCDrop Element

### PR DESCRIPTION
**Pull request by felixheidecke**
Introduced the `<oc-drop />` Element that allows devs to open overlaying containers for little menus for example.

Bildschirmfoto 2019-04-16 um 18 04 53
Bildschirmfoto 2019-04-16 um 18 05 01

### Todo

- [x] Create oc-nav
- [x] Extend oc-nav-item
- [x] Use the above in the oc-drop example
- [x] Fix/enhance documentation

### CODE REVIEW

**Comment by felixheidecke** [comment]
How it difference to oc-menu:

- Can be triggered from outside the component using id, class or attribute selector
- Boundary defaults to the trigger, but it can be placed elsewhere
- The default slot provides a uk-card wrapper for easy filling with "simple" contents
- Supplies an extra, unwrapped slot for any components already bringing their own styles _(can be seen in the example)_
- Can use the whole range of properties uk-drop provides via the options attribute

File: src/elements/OcNavItem.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <li @click="onClick">
-    <a href="#">
+  <li :class="{ 'uk-active': active }">
+    <a href="#" @click.prevent="onClick">
```
Comment (DeepDiver1975) [inline src/elements/OcNavItem.vue]: Naming. Please adjust

File: src/elements/OcNavItem.vue [hunk 2]
```diff
@@ -45,21 +55,20 @@ export default {
...
-      </template>
-    </oc-menu>
-  </template>
-  <script>
-    export default {
-      methods: {
-        onClick(val) {
-          alert(val)
-        }
+  <oc-nav>
```
Comment (DeepDiver1975) [inline src/elements/OcNavItem.vue]: Can you please Model the Filter menu as example Here as Well? Thx

File: src/elements/OCDrop.vue [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div :uk-drop="$_ocDrop_props">
+    <div v-if="$slots.default" class="uk-card uk-card-default uk-card-small uk-card-body">
+      <slot />
+    </div>
+    <slot name="special" />
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: Else is Missing. Or is IT intended that both Slots can be rendered

File: src/elements/OcNavItem.vue (hunk 2 above)
Comment (felixheidecke) [reply src/elements/OcNavItem.vue]: I added the filter menu to oc-drop

File: src/elements/OCDrop.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    /**
+     * CSS selector for the element to be used as toggle. By default, the preceding element is used.
+     **/
+    toggle: {
+      type: String,
+      default: "- *",
+    },
+    /**
+     * Comma separated list of drop trigger behaviour modes
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: please add allowed values to the docs

File: src/elements/OCDrop.vue [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+/**
+ * Position any element in relation to another element.
+ */
+export default {
+  name: "oc-drop",
+  status: "review",
+  release: "1.0.0",
+  props: {
+    /**
+     * CSS selector of the element to maintain the drop's visibility.
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: I don't understand what this is for. Needs better explanation or maybe even an example. THX

File: src/elements/OCDrop.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+     **/
+    mode: {
+      type: String,
+      default: "click",
+      validator: value => {
+        return value.match(/(click|hover)/)
+      },
+    },
+    /**
+     * More options are available
```
Comment (DeepDiver1975) [inline src/elements/OCDrop.vue]: needs more explanation - maybe add link to uikit?

File: src/elements/OCDrop.vue (hunk 4 above)
Comment (DeepDiver1975) [reply src/elements/OCDrop.vue]: taken care off

File: src/elements/OCDrop.vue (hunk 5 above)
Comment (DeepDiver1975) [reply src/elements/OCDrop.vue]: take care off

File: src/elements/OCDrop.vue (hunk 6 above)
Comment (DeepDiver1975) [reply src/elements/OCDrop.vue]: taken care off

**Comment by DeepDiver1975** [comment]
THX @felixheidecke - will merge asap

**Pull request merged**